tests/src/Kernel/PermissionTest.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\collabora_online\Kernel;
+
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\Tests\media\Traits\MediaTypeCreationTrait;
+use Drupal\Tests\user\Traits\UserCreationTrait;
+use Drupal\user\PermissionHandlerInterface;
+
+/**
+ * Tests dynamically created permissions.
+ */
+class PermissionTest extends KernelTestBase {
```

Comment:
Some thoughts about this one @brummbar 
- We could drop the functional test.
- We could move this change to a separate PR, because it is not really required or even related.

For me, the ideal thing would be to have multiple PRs per issue, so that we can isolate changes, but we don't have to open new Jira issues for every technical change.

Reply:
No, no multiple PRs. Our tickets have a scope, we need to stay focused on that. If and when we need multiple PRs, it will be discussed in the ticket or between us case by case.